cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -310,12 +310,19 @@ message TraceBounds {
 message SegYSeismicRequest {
     oneof identifier {
         Identifier seismic = 1;
-        int64 seismic_store_id = 2;
+        int64 seismic_store_id = 2;  // Need to be a data manager or tenant user, not 3rd party, to access by tracestore
     }
+    // This only supports queries to 3D objects.
+    // Deprecated. Use filter instead.
     oneof query {
         Geometry polygon = 3;
         LineBasedRectangle lines = 4;
     }
+    oneof filter {
+        Seismic2dExtent two_dee_extent = 5;   // only valid if the queried object is 2D
+        Seismic3dExtent three_dee_extent = 6; // only valid if the queried object is 3D
```

Comment:
Are the names `Seismic..Extent` a bit misleading in that they can be applied to a seismic store if that's what the identifier specifies?

Reply:
Hmm, good point. Maybe something like `TwoDeeTraceExtent` would be a better name.

Reply:
We could have a separate wrapper for the trace requests or rename the current `Seismic2dExtent` to something generic like `TwoDeeExtent` - https://github.com/cognitedata/seismic-proto/pull/189

Reply:
Maybe we could just include these fields in the `query` oneof and deprecate the `polygon` and `lines` variants?

Reply:
Yeah that might be easier